basic_auth.go

```diff
@@ -95,7 +92,6 @@ func handler(errorHandling promhttp.HandlerErrorHandling) http.Handler {
 		ErrorLog:      log.NewErrorLogger(),
 		ErrorHandling: errorHandling,
 	})
-
```

Comment:
No, we really want to use `handler` function defined there that checks HTTP Basic auth. We should change it to accept registry or gatherer as an argument. Probably it worth to rename that function too to avoid confusion below.

Reply:
@adivinho @DexterHD  We need that code for other exporters. I named it `DefaultMetricsHandler` and moved to `server.go`.